x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.test.tsx

```diff
@@ -4,51 +4,88 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { showAlert } from './AnomalyDetectionSetupLink';
-
-const dataWithJobs = {
-  hasLegacyJobs: false,
-  jobs: [
-    { job_id: 'job1', environment: 'staging' },
-    { job_id: 'job2', environment: 'production' },
-  ],
-};
-const dataWithoutJobs = ({ jobs: [] } as unknown) as any;
-
-describe('#showAlert', () => {
-  describe('when an environment is selected', () => {
-    it('should return true when there are no jobs', () => {
-      const result = showAlert(dataWithoutJobs, 'testing');
-      expect(result).toBe(true);
-    });
-    it('should return true when environment is not included in the jobs', () => {
-      const result = showAlert(dataWithJobs, 'testing');
-      expect(result).toBe(true);
+import React from 'react';
+import { render, fireEvent, wait } from '@testing-library/react';
+import { MissingJobsAlert } from './AnomalyDetectionSetupLink';
+import * as hooks from '../../../../hooks/useFetcher';
+
+async function renderTooltipAnchor({
+  jobs,
+  environment,
+}: {
+  jobs: Array<{ job_id: string; environment: string }>;
+  environment?: string;
+}) {
+  // mock api response
+  jest.spyOn(hooks, 'useFetcher').mockReturnValue({
+    data: { jobs },
+    status: hooks.FETCH_STATUS.SUCCESS,
+    refetch: jest.fn(),
+  });
+
+  const { baseElement, container } = render(
+    <MissingJobsAlert environment={environment} />
+  );
+
+  // hover tooltip anchor if it exists
+  const toolTipAnchor = container.querySelector('.euiToolTipAnchor') as any;
+  if (toolTipAnchor) {
+    fireEvent.mouseOver(toolTipAnchor);
+
+    // wait for tooltip text to be in the DOM
+    await wait(() => {
+      const toolTipText = baseElement.querySelector('.euiToolTipPopover')
+        ?.textContent;
+      expect(toolTipText).not.toBe(undefined);
     });
-    it('should return false when environment is included in the jobs', () => {
-      const result = showAlert(dataWithJobs, 'staging');
-      expect(result).toBe(false);
+  }
+
+  const toolTipText = baseElement.querySelector('.euiToolTipPopover')
+    ?.textContent;
+
+  return { toolTipText, toolTipAnchor };
+}
+
+describe('MissingJobsAlert', () => {
+  it('when no jobs exists - it should show warning', async () => {
+    const { toolTipText, toolTipAnchor } = await renderTooltipAnchor({
+      jobs: [],
     });
+
+    expect(toolTipAnchor).toBeInTheDocument();
+    expect(toolTipText).toBe(
+      'Anomaly detection is not yet enabled. Click to continue setup.'
+    );
   });
 
-  describe('there is no environment selected (All)', () => {
-    it('should return true when there are no jobs', () => {
-      const result = showAlert(dataWithoutJobs, undefined);
-      expect(result).toBe(true);
+  it('when no jobs exists for the selected environment - it should show warning', async () => {
```

Comment:
We should use `describe('when no jobs exists for the selected environment', () => { it('shows a warning', () => >..`